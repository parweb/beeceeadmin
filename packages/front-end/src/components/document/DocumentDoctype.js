import { Button } from '@salesforce/design-system-react';

import { getIcon } from 'helpers';

const DocumentDoctype = ({
  id = null,
  file,
  extension = null,
  size = 1,
  style,
  onClick,
  ...props
}) => {
  const icon = getIcon(extension ? 'file.' + extension : file);

  return (
    <div
      onClick={onClick}
      style={{
        display: 'inline-block',
        cursor: 'pointer',
        ...(size === 2 && {
          padding: '5px',
          margin: '11px'
        })
      }}
    >
      <div
        style={{
          transform: 'scale(' + size + ')',
          position: 'relative',
          display: 'inline-block'
        }}
      >
        <Button
          id={id}
          data-testid="DocumentDoctype"
          style={style}
          {...icon}
          iconSize="large"
          variant="icon"
          {...props}
        />
        {icon.iconName === 'unknown' && (
          <div
            style={{
              position: 'absolute',
              right: '0px',
              left: '0px',
              textAlign: 'center',
              bottom: '6px',
              fontWeight: 400,
              textTransform: 'uppercase',
              color: '#fff',
              fontSize: '7px'
            }}
          >
            {icon.extension}
          </div>
        )}
      </div>
    </div>
  );
};
export default DocumentDoctype;
