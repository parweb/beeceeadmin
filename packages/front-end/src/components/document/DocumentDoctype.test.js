import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import DocumentDoctype from './DocumentDoctype';

describe('<DocumentDoctype />', () => {
  test('render correctly DocumentDoctype component', () => {
    const documentComponent = renderer.create(<DocumentDoctype />);
    const tree = documentComponent.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('renders pdf svg', () => {
    const wrapper = mount(<DocumentDoctype />);

    expect(wrapper.find('svg.slds-button__icon')).toHaveLength(1);
  });

  test('check the value of props', () => {
    const wrapper = mount(
      <DocumentDoctype file="test.call" extention="call" size={2} />
    );

    expect(
      wrapper.find("[data-testid='DocumentDoctype']").props().iconName
    ).toEqual('call');
    expect(
      wrapper.find("[data-testid='DocumentDoctype']").props().iconCategory
    ).toEqual('action');
  });

  test('Check the value of transfrom css  ', () => {
    const wrapper = mount(
      <DocumentDoctype file="test.call" extention="call" size={2} />
    );

    expect(
      wrapper.find("[data-testid='DocumentDoctype']").get(0).props.style
    ).toHaveProperty('transform', 'scale(2)');
  });
});
