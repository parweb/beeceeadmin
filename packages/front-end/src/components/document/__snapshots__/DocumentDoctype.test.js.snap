// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DocumentDoctype /> render correctly DocumentDoctype component 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "position": "relative",
      "transform": "scale(1)",
    }
  }
>
  <button
    className="slds-button slds-button_icon"
    disabled={false}
    id={null}
    onClick={[Function]}
    type="button"
  >
    <svg
      aria-hidden="true"
      className="slds-button__icon slds-button__icon_large"
    >
      <use />
    </svg>
  </button>
  <div
    style={
      Object {
        "bottom": "-9px",
        "color": "#fff",
        "fontSize": "7px",
        "fontWeight": 400,
        "position": "absolute",
        "right": "3px",
        "textTransform": "uppercase",
      }
    }
  />
</div>
`;
